src/platform/plugins/shared/controls/public/control_group/selections_manager.ts

```diff
@@ -40,10 +40,10 @@ export function initSelectionsManager(
         initialTimeslice = controlApi.timeslice$.value;
       }
     });
-    if (initialFilters.length) {
-      filters$.next(initialFilters);
-      unpublishedFilters$.next(initialFilters);
-    }
+
+    filters$.next(initialFilters);
```

Comment:
@nreese I found an issue with not emitting a default [] filter, and after discussing it with @logeekal , he found out that the issue can be solved by setting the default value here. I checked the scenario you mentioned above, and that one is also solved. 🎉 

Would you mind testing this PR again to make sure everything works as before?

Reply:
This is causing an extra fetch for embeddables in dashboards. `filters$.next` should not emit unless there are new filters. In the case where there are no initial filters, it starts as `undefined` and then emits `[]`. `filters$` should not emit if there are no initial filters. I think the best fix is still the previous one where comparators resolve `undefined` and `[]` as equals.